functions/Publish-DbaDacpac.ps1

```diff
@@ -0,0 +1,255 @@
+
+        .NOTES
+            Tags: Migration, Database, Dacpac 
+            Author: Richie lee (@bzzzt_io)
+
+            Website: https://dbatools.io
+            Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+            License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+        .LINK
+            https://dbatools.io/Publish-DbaDacpac
+
+		.EXAMPLE
+			Publish-DbaDacpac -SqlInstance sql2017 -Database WideWorldImporters -Path C:\temp\sql2016-WideWorldImporters.dacpac -PublishXml C:\temp\sql2016-WideWorldImporters-publish.xml 
+			
+			Updates WideWorldImporters on sql2017 from the sql2016-WideWorldImporters.dacpac using the sql2016-WideWorldImporters-publish.xml publish profile
+        
+		.EXAMPLE
+       		Export-DbaDacpac -SqlInstance sql2016 -Database db2 | Publish-DbaDacpac -PublishXml C:\temp\sql2016-db2-publish.xml -Database db1, db2 -SqlInstance sql2017
+		
+			Exports the .dacpac to $home\Documents\sql2016-db2.dacpac then publishes it to the sql2017 server database db2
+	
+  #>
+	[CmdletBinding()]
+	param (
+		[Alias("ServerInstance", "SqlServer")]
+		[DbaInstance[]]$SqlInstance,
+		[Alias("Credential")]
+		[PSCredential]$SqlCredential,
+		[Parameter(Mandatory, ValueFromPipelineByPropertyName)]
+		[string]$Path,
+		[Parameter(Mandatory)]
+		[string]$PublishXml,
+		[Parameter(Mandatory, ValueFromPipelineByPropertyName)]
+		[string[]]$Database,
+		[string[]]$ConnectionString,
+		[switch]$GenerateDeploymentScript,
+		[switch]$GenerateDeploymentReport,
+		[Switch]$ScriptOnly,
+		[string]$OutputPath = "$home\Documents",
+		[switch]$IncludeSqlCmdVars,
+		[switch]$EnableException
+	)
+	
+	begin {
+		if ((Test-Bound -Not -ParameterName SqlInstance) -and (Test-Bound -Not -ParameterName ConnectionString)) {
+			Stop-Function -Message "You must specify either SqlInstance or ConnectionString"
+		}
+		
+		if ((Test-Bound -ParameterName GenerateDeploymentScript) -or (Test-Bound -ParameterName GenerateDeploymentReport)) {
+			$defaultcolumns = 'ComputerName','InstanceName','SqlInstance', 'Database', 'Dacpac', 'PublishXml', 'Result', 'DatabaseScriptPath', 'MasterDbScriptPath', 'DeploymentReport', 'DeployOptions'
+		}
+		else {
+			$defaultcolumns = 'ComputerName', 'InstanceName','SqlInstance', 'Database', 'Dacpac', 'PublishXml', 'Result'
+		}
+		
+		function Get-ServerName ($connstring) {
+			$builder = New-Object System.Data.Common.DbConnectionStringBuilder
+			$builder.set_ConnectionString($connstring)
+			$instance = $builder['data source']
+			
+			if (-not $instance) {
+				$instance = $builder['server']
+			}
+			
+			return $instance.ToString().Replace('\', '-')
+		}
+	}
+	
+	process {
+		if (Test-FunctionInterrupt) { return }
+		
+		if (-not (Test-Path -Path $Path)) {
+			Stop-Function -Message "$Path not found!"
+		}
+		
+		if (-not (Test-Path -Path $PublishXml)) {
+			Stop-Function -Message "$PublishXml not found!"
+		}
+		
+		if ($PipedPublishedProfile.ConnectionString) {
```

Comment:
Should check be removed now that the $pipedPublishPreview variable has is not part of function?

Reply:
oh good point, thank you!